pandas/core/frame.py

```diff
@@ -2025,6 +2022,20 @@ def from_records(
         if columns is not None:
             columns = ensure_index(columns)
 
+        def maybe_reorder(arrays, arr_columns, columns, index):
+            if len(arrays):
+                length = len(arrays[0])
+            else:
+                length = 0
+
+            result_index = None
+            if len(arrays) == 0 and index is None and length == 0:
+                # for backward compat use an object Index instead of RangeIndex
+                result_index = Index([])
+
```

Comment:
can you move this & reorder_arrays to an internals module (and import)?

Reply:
kind of awkward since this is specific to the weird arguments we have in this method.  i agree from_records can use some cleanup, intend to do so in another pass (and have a branch that will give big perf improvements in some structured-ndarray cases)